packages/data-point/README.md

```diff
@@ -1145,18 +1177,125 @@ All entities share a common API (except for [Transform](#transform-entity)).
 
 | Key | Type | Description |
 |:---|:---|:---|
+| *inputType*  | String, [Reducer](#reducers) | type checks the entity's input value, does not mutate value. [Entity Type checking](#entity-type-check). |
 | *before*  | [Reducer](#reducers) | reducer to be resolved **before** the entity resolution |
 | *after*   | [Reducer](#reducers) | reducer to be resolved **after** the entity resolution |
+| *outputType*  | String, [Reducer](#reducers) | type checks the entity's output value, does not mutate value. [Entity Type checking](#entity-type-check). |
 | *error*   | [Reducer](#reducers) | reducer to be resolved in case of an error |
 | *params*  | `Object` | User defined Hash that will be passed to every transform within the context of the transform's execution |
 
-##### <a name="transform-entity">Transform Entity</a>
+
+##### <a name="entity-type-check">Entity Type checking</a>
+
+You may use **inputType** and **outputType** to type check against the values being passed and resolved from an entity. Type checking does not mutate the result. 
+
+**Built in type checks:**
+
+To use built-in type checks you may set the value of **inputType**/**outputType** to: `'string'`, `'number'`, `'boolean'`, `'function'`, `'error'`, `'array'`, or `'object'`. 
+
+<details>
+  <summary>Check if a model outputs an string</summary>
+
+  This example uses a Model Entity, for information on what a model is please go to the [Model Entity](#model-entity) section.
+
+  ```js
+  const dataPoint = DataPoint.create()
+  const helpers = DataPoint.helpers
+
+  dataPoint.addEntities({
+    'model:getName': {
+      value: '$name',
+      outputType: 'string'
+    }
+  })
+
+  const input = {
+    name: 'DataPoint'
+  }
+
+  dataPoint.transform('model:getName', input)
+    .then(acc => {
+      // acc.value -> DataPoint
+    })
+  ```
+
+</details>
+
+**Type Check using Reducer**
+
+To customize type checking you may use a [Reducer](#reducers). If the reducer throws an error, the type check fails.
```

Comment:
it would be helpful to mention that the return value of type check reducers are ignored because of the `.return(acc)` in `typeCheck`